xilem/src/lib.rs

```diff
@@ -335,11 +339,17 @@ impl<W: Widget + FromDynWidget> Pod<W> {
             widget: Box::new(widget),
             id: WidgetId::next(),
             transform: Affine::default(),
+            properties: Properties::new(),
         }
     }
 }
 
 impl<W: Widget + FromDynWidget + ?Sized> Pod<W> {
+    /// Builder-style method to set [`Properties`] on a new widget.
+    pub fn with_props(self, properties: Properties) -> Self {
```

Comment:
It might be nice to warn if the old properties in self weren't empty

Reply:
In practice this can only be used in a new widget, so I don't expect that problem to come up too often.